packages/bitpiewallet/src/adapter.ts

```diff
@@ -0,0 +1,146 @@
+import {
+    EventEmitter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletAdapter,
+    WalletAdapterEvents,
+    WalletDisconnectedError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignatureError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+
+interface BitpieWallet {
+    isBitpieWallet?: boolean;
+    getAccount: () => Promise<string>;
+    signTransaction: (transaction: Transaction) => Promise<Transaction>;
+    signAllTransactions: (transactions: Transaction[]) => Promise<Transaction[]>;
+}
+
+interface BitpieWalletWindow extends Window {
+    solana?: BitpieWallet;
+}
+
+declare const window: BitpieWalletWindow;
+
+export interface BitpieWalletWalletAdapterConfig {
+    pollInterval?: number;
+    pollCount?: number;
+}
+
+export class BitpieWalletWalletAdapter extends EventEmitter<WalletAdapterEvents> implements WalletAdapter {
+    private _connecting: boolean;
+    private _wallet: BitpieWallet | null;
+    private _publicKey: PublicKey | null;
+
+    constructor(config: BitpieWalletWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get ready(): boolean {
+        return !!window.solana?.isBitpieWallet;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet;
+    }
+
+    get autoApprove(): boolean {
+        return false;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+
+            const wallet = window.solana;
+            if (!wallet) throw new WalletNotFoundError();
+            if (!wallet.isBitpieWallet) throw new WalletNotInstalledError();
+
+            // @TODO: handle if popup is blocked
```

Comment:
Can this be removed since Bitpie appears to use a browser extension?

Reply:
yes, I changed this, but `!wallet` is still retained

Reply:
Ah, I just meant the `// @TODO: handle if popup is blocked` comment, but it makes sense that you wouldn't need the `isBitpieWallet` check anymore either since `window.bitpie` is being used.

Reply:
Got it.